pkg/actuators/awsplacementgroup/controller.go

```diff
@@ -0,0 +1,91 @@
+package awsplacementgroup
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	machinev1 "github.com/openshift/api/machine/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	"k8s.io/klog/v2"
+)
+
+// Reconciler reconciles machineSets.
+type Reconciler struct {
+	Client client.Client
+	Log    logr.Logger
+
+	recorder record.EventRecorder
+	scheme   *runtime.Scheme
+}
+
+// SetupWithManager creates a new controller for a manager.
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager, options controller.Options) error {
+	_, err := ctrl.NewControllerManagedBy(mgr).
+		For(&machinev1.AWSPlacementGroup{}).
+		WithOptions(options).
+		Build(r)
+
+	if err != nil {
+		return fmt.Errorf("failed setting up with a controller manager: %w", err)
+	}
+
+	r.recorder = mgr.GetEventRecorderFor("awsplacementgroup-controller")
+	r.scheme = mgr.GetScheme()
+	return nil
+}
+
+// Reconcile implements controller runtime Reconciler interface.
+func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := r.Log.WithValues("awsplacementgroup", req.Name, "namespace", req.Namespace)
```

Comment:
Nit: I think conventionally we would log the name and namespace rather than the kind and namespace
```suggestion
	logger := r.Log.WithValues("name", req.Name, "namespace", req.Namespace)
```